server/game/cycle/stichGranter.js

```diff
@@ -2,98 +2,54 @@ import {GameMode} from '../../../shared/game/gameMode';
 
 const StichGranter = {
     determineWinner: function (mode, trumpfColor, playedCards, players) {
-        let winnerIndex = 0;
-        let winnerCard;
+        const firstPlayedColor = playedCards[0].color;
+        const isModeTrumpf = mode === GameMode.TRUMPF;
+        const trumpfAndFirstPlayedColorCards = playedCards.map(addIndexAndAdjustTrumpfValues)
+                .filter((playedCard) => playedCard.color === firstPlayedColor
+                || (isModeTrumpf && isTrumpf(playedCard)));
+
+        switch (mode) {
+            case GameMode.UNDEUFE:
+                trumpfAndFirstPlayedColorCards.sort((card1, card2) => {
+                    return card1.valueForSorting - card2.valueForSorting
+                });
+                break;
+            case GameMode.OBEABE:
+            case GameMode.TRUMPF:
+                trumpfAndFirstPlayedColorCards.sort((card1, card2) => {
+                    return card2.valueForSorting - card1.valueForSorting
+                });
+                break;
+        }
+
+        return players[trumpfAndFirstPlayedColorCards[0].index];
+
+        function addIndexAndAdjustTrumpfValues(card, index) {
```

Comment:
@r3dDoX I usually declare my functions below as well because it leaves the details below. Although I agree it is maybe a little dirty because they are hoisted (I assume that's why you want to declare them above?), I like that the functions are below and out of the way.